pkg/http/http.go

```diff
@@ -0,0 +1,109 @@
+package http
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
+)
+
+// Client defines an http client object
+type Client struct {
```

Comment:
For me `Client` and `ClientOptions` look pretty the same beside the name. Why is it necessary to have two types?

Reply:
It‘s for having an object for the setter.
In order to be able to abstract the Client for testing there is the func SetOptions.
With that we don’t need to adapt the interface with every new property of the Client.